01_Data/src/layers/sequelize/repository/Authorization.ts

```diff
@@ -140,16 +152,20 @@ export class AuthorizationRepository extends SequelizeRepository<Authorization>
     );
     let savedIdTokenModel: IdToken | undefined;
     if (savedIdTokenId) {
-      savedIdTokenModel = (await this.s.models[IdToken.MODEL_NAME].findOne({
-        where: { id: savedIdTokenId },
-        include: [AdditionalInfo],
-      })) as IdToken;
+      savedIdTokenModel = await this.idToken
+        .readAllByQuery({
+          where: { id: savedIdTokenId },
+          include: [AdditionalInfo],
+        })
+        .then((rows) => rows[0]);
```

Comment:
In order for this to happen, there would need to be a bizarre bug or race condition in the SQL database since savedIdTokenId is a foreign key. I'll add a check regardless.